packages/drivers/odsp-driver/src/localOdspDriver/localOdspDocumentServiceFactory.ts

```diff
@@ -31,7 +30,7 @@ export class LocalOdspDocumentServiceFactory extends OdspDocumentServiceFactoryC
 		const toThrow = new UsageError(
 			`${unsupportedFuncName} is not supported by LocalOdspDocumentServiceFactory`,
 		);
-		this.logger?.sendErrorEvent({ eventName: "UnsupportedUsage" }, toThrow);
+		createOdspLogger().sendErrorEvent({ eventName: "UnsupportedUsage" }, toThrow);
```

Comment:
This will create a new logger every time we throw an exception. This doesn't seem ideal. Is there a reason we removed the logger property?

Reply:
The logger was a deprecated type, ITelemetryLogger. There is internal used local service should be ok to use `Ext` type logger. It needs a logger when there is an error thrown there. I think we don't need to create it each time just when error occurred in this case.

Reply:
Introducing a `logger` property on the parent class was causing a conflict here because of mismatching types (even though they are both private; thank you JS). But the new property wasn't necessary in the parent class, I got rid of it and that let me keep this one with a logger that is only instantiated once.

Reply:
Also of note, I think these log messages might not have been working as intended. I'm not sure there is a guarantee that `createDocumentServiceCore()` which sets the logger was called before `createContainer()` which wants to use it (or the functions set in the constructor that also want to use it). Instead of using a logger passed in during `createDocumentServiceCore()` I just created it in this class with the same namespace that the logger it could receive.